jhipster/generator-jhipster
# JHipster CLI

- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

@jhipster/developers this is the first version of the JHipster CLI

Please test it and provide feedback to me.

To test this locally
1.  Pull this branch
```
git checkout -b deepu105-jh-cli
git pull https://github.com/deepu105/generator-jhipster.git jh-cli
```
2. Run `npm install` or yarn equivalent
3. Run `npm install -g` or yarn equivalent
4. Run `npm link` or yarn equivalent
5. The CLI should be installed in path now. Create a new folder somewhere and run below commands to test (please note to disable the jhipster zsh/fisherman plugin if you have it setup)

- The initial version basically maps all our existing `yo jhipster:*` commands to `jh *`. All flags supported by each subgen can be passed in the same way for eg: `jh entity testEntity --skip-client`. the `--help` flag lists all possible arguments and flags
- A `-d` flag can be passed to enable debug logs
- `jh --help` prints all available commands and options
- `jh` runs the default command equivalent to `yo jhipster` supports all the flags present
- `jh --version` prints version
